acceptance/tests/environment/negative/agent_run_should_fail_if_env_unreadable.rb

```diff
@@ -0,0 +1,57 @@
+test_name "C97899 - Agent run should fail if environment is unreadable" do
+
+  testdir = ''
+  env_path = ''
+  test_env = ''
+
+  step 'setup environments' do
+    testdir = create_tmpdir_for_user master, 'c97899_unreadable_envdir'
+    env_path = "#{testdir}/environments"
+    test_env = "#{env_path}/testing"
+
+    apply_manifest_on(master, <<-MANIFEST, :catch_failures => true)
+      File {
+        ensure => directory,
+        mode => "0770",
+        owner => #{master.puppet['user']},
+        group => #{master.puppet['group']},
+      }
+      file {
+        '#{env_path}':;
+        '#{env_path}/production':;
+        '#{test_env}':;
+        '#{test_env}/manifests':;
+        '#{test_env}/modules':;
+      }
+      file { '#{test_env}/manifests/site.pp':
+        ensure => file,
+        mode => "0640",
+        content => 'node default { notify { "Hello agent": } }',
+      }
+    MANIFEST
+  end
+
+  step 'change permissions of envdir to 644' do
+    on(master, "chmod 644 #{test_env}")
+  end
+
+  step 'verify environment fails with puppet agent run' do
+    master_opts = {
+      'main' => {
+        'environmentpath' => env_path,
+      }
+    }
+    with_puppet_running_on master, master_opts, testdir do
```

Comment:
If run with the `ci:test:aio` rake task, the `with_puppet_running_on` uses the deployed puppetserver rather than a passenger based master. We could throttle the test to only run in the context of a `type == aio` run. If we want this test to be added to the regression suite, I would recommend that we run it without such a restriction.